File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> umap;
        vector<int> hash;
        for (const auto &str: strs) {
            hash.assign(26, 0);
            for (const auto &x: str) {
                hash[tolower(x) - 'a']++;
            }
            string encode = "";
            for (const auto &x: hash) encode += x;
            umap[encode].push_back(str);
        }
        vector<vector<string>> res;
        for (const auto &x: umap) res.push_back(x.second);
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.cs
public class Solution {
    public IList<IList<string>> GroupAnagrams(string[] strs) {
        var anagrams = new Dictionary<string, List<string>>();
        foreach (string str in strs) {
            char[] x = str.ToCharArray();
            Array.Sort(x);
            string key = new string(x);
            if (!anagrams.ContainsKey(key)) {
                anagrams[key] = new List<string>();
            }
            anagrams[key].Add(str);
        }
        var res = new List<IList<string>>();
        foreach (var group in anagrams.Values) {
            res.Add(group);
        }
        
        return res;
    }
}
